src/processors.zig

```diff
@@ -181,6 +179,46 @@ pub const CrossPageLinkProcessor = struct {
             }
         }
     }
+    fn parseAltText(
+        pctx: anytype,
+        alt_text: []const u8,
+        maybe_scale: ?[]const u8,
+    ) !void {
+        if (maybe_scale) |scale| {
+            if (parseResolutionFromAlt(scale)) |res| {
+                try pctx.out.print("alt=\"{s}\"", .{alt_text});
+                try pctx.out.print("width=\"{s}\"", .{res[0].?});
+                if (res[1] != null) try pctx.out.print("height=\"{s}\"", .{res[1].?});
+                return;
+            }
+
+            // Third position exists but is not a resolution
+            // Treat scale as a literal value and reconstruct what would have been the alt text
+            try pctx.out.print("alt=\"{s}|{s}\"", .{ alt_text, scale });
+            return;
+        }
+
+        if (parseResolutionFromAlt(alt_text)) |res| {
+            try pctx.out.print("width=\"{s}\"", .{res[0].?});
+            if (res[1] != null) try pctx.out.print("height=\"{s}\"", .{res[1].?});
+            return;
+        }
+
+        try pctx.out.print("alt=\"{s}\"", .{alt_text});
+    }
+
+    fn parseResolutionFromAlt(alt: []const u8) ?[2]?[]const u8 {
```

Comment:
nitpick: i think (haven't checked myself) could be modeled cleaner instead of `?[2]?[]const u8`, like this

```zig
type ExplicitResolution = struct {
    width: isize
    height: ?isize
};
type AltResult = union(enum) {
    resolution: ExplicitResolution,
    alt_text: []const u8
};
```

then `switch(result) { ... }`

Reply:
Where should I put such types? I tried to keep it relatively self contained and I'm seeing no types for Processors I can use as reference.

Reply:
you can declare types inside of structs, example:

```
    const Self = @This();
```

Self is a type, `@This();` returns a reference to the type being currently declared at that syntax level

Reply:
Done, got tripped up that I had to write declare it per `const` and not `type` so it can be part of the struct.